### PR TITLE
1.0

### PR DESCRIPTION
**Pull request by ChrisYammine**
Breaking changes!

Modifying all return values from functions to match what is idiomatic Elixir. For example:

```elixir
# Successful
{:ok, response}

# Error
{:error, %ExTrello.Error{code: 404, message: "list not found"}}

# Connection Error
{:connection_error, %ExTrello.ConnectionError{reason: "nxdomain", message: "Connection error."}}
```

This should make a lot more sense to users as they will no longer have to handle any raised exceptions. 

If I missed any changes to documentation [please shoot an email over to me](mailto:cyammine91@gmail.com) or open an issue on this repo 😄

### CODE REVIEW

File: lib/ex_trello.ex [hunk 1]
```diff
@@ -224,17 +222,17 @@ defmodule ExTrello do
...
-      # board = %ExTrello.Model.Board{id: "57663306e4b15193fcc97483" blah blah blah}
+      # {:ok, board} = %ExTrello.Model.Board{id: "57663306e4b15193fcc97483" blah blah blah}
       board |> ExTrello.board_cards(members: true)
 
   ## Reference
   https://developers.trello.com/advanced-reference/board#get-1-boards-board-id-cards
   """
 
-  @spec board_cards(String.t | ExTrello.Model.Board.t, Keyword.t) :: [ExTrello.Model.Card.t] | []
+  @spec board_cards(String.t | ExTrello.Model.Board.t, Keyword.t) :: {:ok, [ExTrello.Model.Card.t]}
```
Comment (msanterre) [inline lib/ex_trello.ex]: Shouldn't you also put `{:error, etc}` in the return spec?

File: README.md [hunk 2]
```diff
@@ -4,6 +4,34 @@ A library for interfacing with the Trello API.
...
+try do
+  ExTrello.get("potato/2")
+rescue
+  %ExTrello.Error{code: code, message: message} -> IO.puts "ERROR[#{code}] - #{message}"
+end
+
+# New and fantastic
+case ExTrello.get("potato/2") do
+  {:ok, response} -> response
+  {:error, code, message} -> IO.puts "ERROR[#{code}] - #{message}"
```
Comment (ChrisYammine) [inline README.md]: Perhaps just `{:error, %ExTrello.Error{}}` instead? The only time an error is thrown is when communication is happening over HTTP (except for the missing credentials....which should maybe raise?)

File: lib/ex_trello.ex (hunk 1 above)
Comment (ChrisYammine) [inline lib/ex_trello.ex]: Good point.

File: README.md (hunk 2 above)
Comment (msanterre) [inline README.md]: I think none of them should raise, people should be forced to handle all kinds of errors.

I think removing the code would work. You could possibly add the code inside of the struct for extra debugging info though. Also provides some context and extendability.

Comment (ChrisYammine) [inline README.md]: Done 👍

File: lib/ex_trello.ex (hunk 1 above)
Comment (ChrisYammine) [inline lib/ex_trello.ex]: Done 👍

**Comment by msanterre** [comment]
Eggcellent 👍

**Pull request merged**